source/iterator/iterator.go

```diff
@@ -35,8 +35,9 @@ type SheetsIterator struct {
 	rowOffset int64
 	tomb      *tomb.Tomb
 	ticker    *time.Ticker
-	caches    chan []sdk.Record
-	buffer    chan sdk.Record
+	// haris: what's the difference between caches and buffer?
+	caches chan []sdk.Record
+	buffer chan sdk.Record
```

Comment:
When the iterator fetches a slice of records from Google Sheets, why not put them individually into `caches`?

Reply:
@boldCoder bump : )

Reply:
@hariso That will block our iterator till all the existing records have been read. This is the same functionality across Redis, zendesk, sheets, and s3 connector. 
Comment from @maha: https://github.com/gopherslab/conduit-connector-redis/pull/8/files#diff-e24a48fcc6c1fe4841b2397219b35a1f4205ffd25060a0d2189be5e5a5077b77R147

Reply:
@amansardana Hmm, but why? We would be taking a slice of records from the Sheets API, putting them one by one into the `caches`  channel. The iterator reads from the channel as long as there's something to read. If there isn't anything after some time, it returns an ErrBackoffRetry. Or maybe I'm missing something?

@maha-hajja do you maybe know more about why we do this?

Reply:
@hariso We can do that if the current functionality is not desired.

Reply:
I've pinged Maha about this. If needed, I'm fine with merging this as it is, and then simplifying later (if possible at all).

Reply:
@hariso @amansardana this is how I think about this specific function:
if we insert them one by one into the cache then this would block the goroutine from doing anything else and wait for the whole cache to be consumed then start fetching new records and sort them {which may take time, and conduit might try reading throughout this time}.. on the other hand, the goroutine could strat fetching new records and prepare them while conduit is still reading from the cache.. so why wait for the cache to get empty then start getting new records if you can optimize the time?
@hariso of course the way you're talking about works! but it's a matter of what do we think would be better. what do you think?